pkg/controller/deployment/deployment_controller.go

```diff
@@ -959,6 +957,10 @@ func (dc *DeploymentController) calculateStatus(allRCs []*api.ReplicationControl
 }
 
 func (dc *DeploymentController) scaleRCAndRecordEvent(rc *api.ReplicationController, newScale int, deployment extensions.Deployment) (*api.ReplicationController, error) {
+	// No need to scale
+	if rc.Spec.Replicas == newScale {
+		return rc, nil
```

Comment:
Made `scaleRCAndRecordEvent` return a `bool` for whether the scale event happen or not.